### ISSUE TITLE
The BOM is readed as a value in the content

### ISSUE THREAD

**Issue by Chimrod** [Chris00/ocaml-csv#36]
Excel register the CSV encoded in UTF-8 by adding a [Byte Order Mark](https://en.wikipedia.org/wiki/Byte_order_mark) in the very beginning of the file.

The library read thoses bytes and considered them as a part of the content, this can raise strange errors when the first cell is quoted, as in this minimal example:

[bom.csv](https://github.com/Chris00/ocaml-csv/files/8870215/bom.csv)
 
```ocaml
# Csv.load "bom.csv" ~separator:';';;
Exception: Csv.Failure (2, 1, "Bad '\"' in quoted field").
```

I open the issue because I can’t guess if thoses bytes are presents when I stream a channel: I can discard the first 3 bytes if I a BOM is present, but I have no way to check for it, and revert the channel to the initial state if there is no BOM. So I think this should be handled in the library itself.

**Comment by Chimrod** [comment]
Yes, I would be happy to contribute :)

Just give me some time in order to go into the code and I will open a PR

**Comment by Chimrod** [comment]
I’ve started the branch here : https://github.com/Chimrod/ocaml-csv/tree/bom

I’ll had a test before requesting a merge.

**Comment by SGrondin** [comment]
Fixed in #41
